x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -0,0 +1,160 @@
+
+jest.mock('node-fetch');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mock;
+const { Response } = jest.requireActual('node-fetch');
+
+const responseMock = {
+  ok: jest.fn(),
+  notFound: jest.fn(),
+  customError: jest.fn(),
+};
+const KibanaAuthHeader = 'Basic 123';
+
+interface IMockResponse {
+  results: object[];
+  meta?: {
+    page?: {
+      total_results: number;
+    };
+  };
+}
+
+describe('createEnterpriseSearchRequestHandler', () => {
+  beforeEach(() => {
+    responseMock.ok.mockClear();
+    responseMock.notFound.mockClear();
+    responseMock.customError.mockClear();
+    fetchMock.mockReset();
+  });
+  it('makes an API call and returns the response', async () => {
+    const appSearchAPIResponseBody = {
+      results: [{ name: 'engine1' }],
+      meta: { page: { total_results: 1 } },
+    };
+
+    AppSearchAPI.mockReturn(appSearchAPIResponseBody);
+
+    const requestHandler = createEnterpriseSearchRequestHandler({
+      config: mockConfig,
+      log: mockLogger,
+      path: '/as/credentials/collection',
+    });
+
+    await makeAPICall(requestHandler, {
+      query: {
+        type: 'indexed',
+        pageIndex: 1,
+      },
+    });
+
+    AppSearchAPI.shouldHaveBeenCalledWith(
+      'http://localhost:3002/as/credentials/collection?type=indexed&pageIndex=1',
+      {
+        headers: { Authorization: KibanaAuthHeader },
+      }
+    );
+
+    expect(responseMock.ok).toHaveBeenCalledWith({
+      body: appSearchAPIResponseBody,
+    });
+  });
+
+  describe('when an API request fails', () => {
+    it('should return 404 with a message', async () => {
+      AppSearchAPI.mockReturnError();
+
+      const requestHandler = createEnterpriseSearchRequestHandler({
+        config: mockConfig,
+        log: mockLogger,
+        path: '/as/credentials/collection',
+      });
+
+      await makeAPICall(requestHandler);
+
+      AppSearchAPI.shouldHaveBeenCalledWith(`http://localhost:3002/as/credentials/collection?`, {
+        headers: { Authorization: KibanaAuthHeader },
+      });
+
+      expect(responseMock.customError).toHaveBeenCalledWith({
+        body: 'Error connecting or fetching data from Enterprise Search',
+        statusCode: 502,
+      });
+    });
+  });
+
+  describe('when `hasValidData` fails', () => {
+    it('should return 404 with a message', async () => {
+      const appSearchAPIResponseBody = {
+        foo: 'bar',
+      };
+
+      AppSearchAPI.mockReturn(appSearchAPIResponseBody);
+
+      const requestHandler = createEnterpriseSearchRequestHandler({
+        config: mockConfig,
+        log: mockLogger,
+        path: '/as/credentials/collection',
+        hasValidData: (body?: IMockResponse) => {
+          return (
+            Array.isArray(body?.results) && typeof body?.meta?.page?.total_results === 'number'
+          );
+        },
```

Comment:
Super minor nit, we can lose a few extra lines here w/ arrow function syntax:

```suggestion
        hasValidData: (body?: any) =>
          Array.isArray(body?.results) && typeof body?.meta?.page?.total_results === 'number',
```

Also threw in the `any` change from above and seems to work OK